server/grails-app/controllers/org/gokb/OaiController.groovy

```diff
@@ -198,9 +198,12 @@ class OaiController {
       def oid = params.identifier
       def record = null
       Boolean returnAttrs = true
-      Boolean cachedPackageResponse = (result.oaiConfig.id == 'packages' && grailsApplication.config.getProperty('gokb.packageOaiCaching.enabled', Boolean, false))
+      Boolean cachedPackageResponse = (options.oaiConfig.id == 'packages' && grailsApplication.config.getProperty('gokb.packageOaiCaching.enabled', Boolean, false))
       def request_map = params
-      def legalClassNames = (result.className == 'org.gokb.cred.TitleInstance' ? ['org.gokb.cred.TitleInstance', 'org.gokb.cred.BookInstance', 'org.gokb.cred.JournalInstance', 'org.gokb.cred.DatabaseInstance', 'org.gokb.cred.OtherInstance'] : [result.className])
+      def legalClassNames = (options.className == 'org.gokb.cred.TitleInstance' ?
+                                ['org.gokb.cred.TitleInstance', 'org.gokb.cred.BookInstance', 'org.gokb.cred.JournalInstance', 'org.gokb.cred.DatabaseInstance', 'org.gokb.cred.OtherInstance'] :
+                                [result.className])
```

Comment:
It looks like `result.className` here might need to be changed to `options.className`